oat-sa/extension-tao-testqti
# [NCC-399] Backport TR-4008 session state restoration

# Introduction

This Backport is intended to solve the issue [NCC-399](https://oat-sa.atlassian.net/browse/NCC-399).

Current Implementation:
The states are being archived in a synchronous manner and can not be configured to be saved on s3
With the backport:
The states will be archived in an asynchronous manner using the tao task queue and saved on s3

# How to test

1. Setup the `/path/to/tao/nccer/config/generis/event.conf.php`:
```
        'oat\\taoQtiTest\\models\\event\\AfterAssessmentTestSessionClosedEvent' => array(
            array(
                'oat\\taoQtiTest\\models\\QtiTestListenerService',
                'archiveState'
            )
        ),
```
2. Take a test
3. check your `stateBackup` file system. You should have the states saved there:
<img width="1369" alt="Screenshot 2022-08-10 at 14 43 22" src="https://user-images.githubusercontent.com/11900046/183904245-d4b2bf40-c9f6-4698-a45c-a4fbd9f68634.png">
